File: src/main/java/me/lovesasuna/lanzou/file/FileItem.java
package me.lovesasuna.lanzou.file;

import java.io.BufferedReader;
import java.net.URL;

/**
 * @author LovesAsuna
 * @date 2020/8/22 10:11
 **/
public abstract class FileItem extends Item {
    protected final String MAGIC = "_c_c";
    URL downloadableUrl;
    protected String icon;
    protected String name;
    protected String size;
    protected String time;
}

File: src/main/kotlin/me/lovesasuna/lanzou/file/FileItem.kt
package me.lovesasuna.lanzou.file

import java.net.URL

/**
 * @author LovesAsuna
 * @date 2020/8/22 10:11
 */
abstract class FileItem : Item() {
    protected val MAGIC = "_c_c"
    lateinit var downloadableUrl: URL
        protected set
    lateinit var icon: String
    lateinit var name: String
    lateinit var size: String
    lateinit var time: String
}